### PR TITLE
Refactor SonarAnalysisContext - Bump coverage

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Bump coverage for #6532

When this branch targetted master, to see coverage for all commits, this was the result:

We're on 99.6% of coverage.
All really-changed-code is covered.
I did not try to bump coverage for any rule.

image

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarSyntaxTreeAnalysisContextTest.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public void Properties_ArePropagated()
+    {
+        var cancel = new CancellationToken(true);
+        var (tree, model) = TestHelper.CompileCS("// Nothing to see here");
+        var context = new SyntaxTreeAnalysisContext(tree, null, _ => { }, _ => true, cancel);
+        var sut = new SonarSyntaxTreeAnalysisContext(TestHelper.CreateSonarAnalysisContext(), context, model.Compilation);
+
+        sut.Tree.Should().Be(tree);
+        sut.Compilation.Should().Be(model.Compilation);
+        sut.Cancel.Should().Be(cancel);
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarSyntaxTreeAnalysisContextTest.cs:49]: Options are missing also from here

File: analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SonarAnalysisContextExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using VB = vbnet::SonarAnalyzer.Extensions.SonarAnalysisContextExtensions;
+
+namespace SonarAnalyzer.UnitTest.Extensions;
+
+[TestClass]
+public class SonarAnalysisContextExtensions
+{
+    private static readonly DiagnosticDescriptor DummyMainDescriptor = TestHelper.CreateDescriptor("Sxxxx", DiagnosticDescriptorFactory.MainSourceScopeTag);
+
+    public TestContext TestContext { get; set; }
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SonarAnalysisContextExtensionsTest.cs:37]: not used

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarCompilationAnalysisContextTest.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        var analysisContext = new SonarAnalysisContext(Mock.Of<RoslynAnalysisContext>(), Enumerable.Empty<DiagnosticDescriptor>());
+        var cancel = new CancellationToken(true);
+        var (tree, model) = TestHelper.CompileCS("// Nothing to see here");
+        var context = new CompilationAnalysisContext(model.Compilation, null, _ => { }, _ => true, cancel);
+        var sut = new SonarCompilationAnalysisContext(analysisContext, context);
+
+        sut.Cancel.Should().Be(cancel);
+        sut.Tree.Should().Be(tree);
+        sut.Compilation.Should().Be(model.Compilation);
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarCompilationAnalysisContextTest.cs:41]: I was adding mostly those that were not covered.. I'll add them all

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! Tests will be automatically fixed after rebase.

**Pull request merged**